UC-Berkeley-Extension/global
# Angular

## Asana Ticket(s)
- See ticket [Do we want to use AngularJS in Drupal](https://app.asana.com/0/1174970560648905/1194018677557889/f).

## Description
- Adds the [PDB](https://www.drupal.org/project/pdb/) module for Angular blocks.
- Enables a sample block.

## Testing instructions

* Checkout this branch
* From project root, run `composer install`
* Check that the `pdb` module downloaded properly.
* SSH into DDEV `ddev ssh`
* Run `../scripts/setup.sh` to re-install Drupal
* Go to https://global.ddev.site/ 
 * Observe the `Voices Post List` block is displayed on the homepage and has content.
* Login using `drush uli`
* Observe that the `bglobal_angular` module is enabled.
* Go to https://global.ddev.site/admin/structure/block
* Observe that the `Angular: Voices Post List` block is enabled in the Content section.
* Edit the block at https://global.ddev.site/admin/structure/block/manage/angularvoicespostlist?destination=/admin/structure/block/list/bglobal
* Observe that you can edit the following values:
  - post
  - tag
  - program
* Set post to `1926` and save the block
* Go to the homepage
* Observe that the block only shows post 1926.
 
## Screenshots

![Screen Shot 2020-09-29 at 5 08 59 PM](https://user-images.githubusercontent.com/360988/94615948-97234680-0276-11eb-9fe3-50d230b6dca7.png)


## Remaining tasks

- [ ] Figure out how to compile .ts to .js using `tsc` so we don't have to use development mode of the module
- [x] Figure out how to get the settings from two instances of the block on the same page.

This task can be done as a follow-up.
